app/store/ephemeral_store.ts

```diff
@@ -25,6 +32,66 @@ class EphemeralStore {
     private notificationTapped = false;
     private enablingCRT = false;
 
+    // Ephemeral control for out of order websocket events
+    addEditingPost = (serverUrl: string, post: Post) => {
+        if (this.websocketRemovingPost[serverUrl]?.has(post.id)) {
+            return;
+        }
+
+        const lastEdit = this.websocketEditingPost[serverUrl]?.[post.id];
+        if (lastEdit && post.edit_at < lastEdit.post.update_at) {
+            return;
+        }
+
+        if (!this.websocketEditingPost[serverUrl]) {
+            this.websocketEditingPost[serverUrl] = {};
+        }
+        const serverEditing = this.websocketEditingPost[serverUrl]!;
+
+        if (lastEdit?.timeout) {
+            clearTimeout(lastEdit.timeout);
+        }
+
+        const timeout = setTimeout(() => {
+            delete serverEditing[post.id];
+        }, TIME_TO_CLEAR_WEBSOCKET_ACTIONS);
+
+        serverEditing[post.id] = {post, timeout};
+    };
+
+    addRemovingPost = (serverUrl: string, postId: string) => {
+        if (this.websocketRemovingPost[serverUrl]?.has(postId)) {
+            return;
+        }
+
+        if (this.websocketEditingPost[serverUrl]?.[postId]) {
+            clearTimeout(this.websocketEditingPost[serverUrl]![postId]!.timeout);
```

Comment:
shouldn't we check that timeout is not undefined here?

Reply:
Timeout should always be defined (we always create the object with a post and a timeout), and everything to the left of `.timeout` must be defined because if not, it wouldn't go through the `if` clause.